File: api/omp4m/codegen/lib/ompGetNumProcs/ompGetNumProcs.c
#include "ompGetNumProcs.h"
#include "omp.h"

int ompGetNumProcs(void)
{
  return omp_get_num_procs();
}

void ompGetNumProcs_initialize(void)
{
}

void ompGetNumProcs_terminate(void)
{
}

File: api/omp4m/ompGetNumProcs.m
function n = ompGetNumProcs %#codegen
%Returns the number of processors that are available to the program.

%#codegen -args {}

n = int32(1);
if ~strcmp(coder.target,'rtw'); return; end

coder.cinclude('omp.h');
coder.inline('always');
coder.allowpcode('plain')

n = coder.ceval('omp_get_num_procs');
